docs/reference/docs/concurrency-control.asciidoc

```diff
@@ -0,0 +1,111 @@
+by the primary shard that coordinates that change. The sequence number is increased
+with each operation and thus newer operations are guaranteed to have a higher sequence
+number. Elasticsearch can then use the sequence number of operations to make sure they
+never override a newer document version is never overridden by a change that has a 
+smaller sequence number assigned to it. 
+
+For example, the following indexing command will create a document and assign it its 
+initial sequence number and primary term:
+
+[source,js]
+--------------------------------------------------
+PUT products/_doc/1567
+{
+    "product" : "r2d2",
+    "details" : "A resourceful astromech droid"
+}
+--------------------------------------------------
+// CONSOLE
+
+You can see the assigned sequence number and primary term in the 
+the `_seq_no` and `_primary_term` fields of the response:
+
+[source,js]
+--------------------------------------------------
+{
+    "_shards" : {
+        "total" : 2,
+        "failed" : 0,
+        "successful" : 1
+    },
+    "_index" : "products",
+    "_type" : "_doc",
+    "_id" : "1567",
+    "_version" : 1,
+    "_seq_no" : 362,
+    "_primary_term" : 2,
+    "result" : "created"
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"_seq_no" : \d+/"_seq_no" : $body._seq_no/ s/"_primary_term" : 2/"_primary_term" : $body._primary_term/]
+
+
+Elasticsearch keeps tracks of the sequence number and primary of the last
+operation to have changed each of the document it stores. These are returned
+as part of the response of the <<docs-get,GET API>>:
+
+[source,js]
+--------------------------------------------------
+GET products/_doc/1567
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+returns:
+
+[source,js]
+--------------------------------------------------
+{
+    "_index" : "products",
+    "_type" : "_doc",
+    "_id" : "1567",
+    "_version" : 1,
+    "_seq_no" : 362,
+    "_primary_term" : 2,
+    "found": true,
+    "_source" : {
+        "product" : "r2d2",
+        "details" : "A resourceful astromech droid"
+    }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"_seq_no" : \d+/"_seq_no" : $body._seq_no/ s/"_primary_term" : 2/"_primary_term" : $body._primary_term/]
+
+
+Note: The <<search-search,Search API>> can return the `_seq_no` and `_primary_term`
+for each search hit by requesting the `_seq_no` and `_primary_term` <<search-request-docvalue-fields,Doc Value Fields>>.
+
+The sequence number and the primary term uniquely identify a change. By noting down 
+the sequence number and primary term return, you can make sure to only change the 
+document if no other change was made to it since you retrieved it. This
+is done by setting the `if_seq_no` and `if_primary_term` parameters of either the
+<<docs-index_,Index API>> or the <<docs-delete,Delete API>>. 
+
+For example, the following indexing call, will make sure to add a tag to the
```

Comment:
no comma needed:

For example, the following indexing call will make sure to add a tag to the